### ISSUE TITLE
Supervised clustering on meta data using UMAP

### ISSUE THREAD

**Issue by solyris** [satijalab/seurat#1463]
HI, I am trying to use UMAP to perform a supervised clustering on a metadata which I have. For each single-cell barcode, we have assigned a cluster ID (1-40) curated from another set of data related to the gene expression data from 10X which we have created a Seurat object from the data. What we want to do is cluster the gene expression data based on this 40 cluster.

I think UMAP-learn has something which I need but I am not sure how to analyse my Seurat object with that. Is there anyway to do the same as "Using Labels to Separate Classes (Supervised UMAP)" found in the link below for Seurat objects?
In particular, this is the function I think is needed "umap.UMAP().fit_transform(data, y=target)"
https://umap-learn.readthedocs.io/en/latest/supervised.html

Reagrds
Solyris

**Comment by andrewwbutler** [comment]
Hi,

Seurat doesn't currently support that functionality of UMAP. You can try our version of transfering labels from an annotated reference ([vignette here](https://satijalab.org/seurat/pancreas_integration_label_transfer.html)).
